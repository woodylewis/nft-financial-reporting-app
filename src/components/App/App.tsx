import * as React from 'react';
import Container from '@mui/material/Container';
import { Store } from '../Store/Store';

import Gateway from '../Gateway/Gateway';

const App = () => {
  return (
    <Container maxWidth='md'>
      <Store>
        <Gateway />
      </Store>
    </Container>
  );
};

export default App;
