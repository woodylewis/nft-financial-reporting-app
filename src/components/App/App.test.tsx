import * as React from 'react';
import { render, cleanup, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import App from './App';

describe('App component', () => {
  afterEach(cleanup);

  it('should have a link to network', () => {
    render(<App />);
    expect(screen.getByRole('link')).toHaveAttribute('href', '/network');
  });

  it('clicking link should navigate to network', async () => {
    render(<App />);
    await fireEvent.click(screen.getByRole('link'));
    expect(screen.getByText('NETWORK')).toBeTruthy;
  });

  it('should match snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment).toMatchSnapshot();
  });
});
