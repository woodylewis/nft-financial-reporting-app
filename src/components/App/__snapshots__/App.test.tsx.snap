// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match snapshot 1`] = `[Function]`;
